### PR TITLE
Analysis AOI tool

### PR DESCRIPTION
**Pull request by danielfdsilva**
Adds the analysis aoi selection tool to the analysis define page.

Opening as Draft for review, but will rebase when header related prs are merged. There was code needed on those branches for this to work.

### CODE REVIEW

**Review by nerik — APPROVED** [review]
Looking good!

We need to use the URL as a single source of truth, instead of maintaining a copy of the state in the URL. Which I will address in <a href="https://github.com/NASA-IMPACT/delta-ui/pull/284">this PR</a> using react-router's `useSearchParams`

**Comment by danielfdsilva** [comment]
@nerik Actually in this case we don't really use the url as a source of truth.

When we start a fresh analysis the url is empty. As we change the parameters, these should be stored in the internal state and then stringified for the Save button url (the url for the analysis results page).

If we are refining an analysis the parameters in the url (which got there from the analysis results refine button), are just the starting point, and are also how we know that we're refining an analysis instead of starting fresh.

Basically in the case of the analysis the url is read-only, we don't need to write to it.

Happy to chat about this if it is not clear or you see a different need.

File: app/scripts/components/analysis/define/aoi-selector.tsx [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+export default function AoiSelector(props: AoiSelectorProps) {
+  const { onAoiEvent, qsFeature, aoiDrawState } = props;
+  const { selected, drawing, feature } = aoiDrawState;
+  const mapRef = useRef<MapboxMapRef>();
+
+  // Technical debt
+  // Despite the query parameters support for multiple features on the aoi, the
+  // AOI drawing tool only supports one.
+  // Keeping just the first one.
+  const polygon: Feature<Polygon> | null = useDeepCompareMemo(() => {
```
[the remark this answers is not in the document]
Comment (danielfdsilva) [reply app/scripts/components/analysis/define/aoi-selector.tsx:63]: That is true, but it would involve passing the raw string around, which I think it is confusing.
Nevertheless, qsFeature _should_ be a stable object (or possible to be made one), but I need to double check.

**Comment by danielfdsilva** [comment]
@nerik turns out the `useDeepCompareMemo` in `aoi-selector.tsx` was not needed. `qsFeature` is stable.
With the merge of the header to main, I rebased this branch before marking the PR for review.

**Review by nerik — APPROVED** [review]
🚀
